integration-cli/docker_cli_exec_test.go

```diff
@@ -349,9 +348,16 @@ func (s *DockerSuite) TestInspectExecID(c *check.C) {
 		c.Fatalf("failed to get the exec id: %v", err)
 	}
 
-	// End the exec by closing its stdin, and wait for it to end
-	execStdin.Close()
-	cmd.Wait()
+	// End the exec by creating the missing file
+	err = exec.Command(dockerBinary, "exec", "-i", id,
+		"sh", "-c", "touch /tmp/execid1").Run()
+
+	if err != nil {
+		c.Fatalf("failed to run the 2nd exec cmd: %q", err)
+	}
+
+	// Give the first exec time to detect the file
+	time.Sleep(2 * time.Second)
```

Comment:
Instead can save the cmd for the first exec to a var, then call cmd.Wait() here.